.github/workflows/build-snap.yml

```
@@ -0,0 +1,38 @@
+name: Snap Builder
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main, IENG-* ]
+  # Allow manual trigger
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: 
+          - arm64
+    steps:
+        
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setub QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: ${{ matrix.platform }}
+
+      - name: Build snap for arm64
+        uses: diddlesnaps/snapcraft-multiarch-action@v1
+        id: build
+        with:
+          architecture: ${{ matrix.platform }}
+      
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: matter-pi-gpio-commander_${{ github.run_number}}_1.1.0_arm64.snap
```

```suggestion
          name: matter-pi-gpio-commander_${{ github.run_number}}_arm64.snap
```
Best to exclude hardcoded version number here for better maintainability.